ui/admin/app/components/form/credential/index.js

```diff
@@ -5,7 +5,26 @@
 
 import Component from '@glimmer/component';
 import { service } from '@ember/service';
-import { TYPES_CREDENTIAL } from 'api/models/credential';
+import {
+  TYPES_CREDENTIAL,
+  TYPE_CREDENTIAL_USERNAME_PASSWORD,
+  TYPE_CREDENTIAL_SSH_PRIVATE_KEY,
+  TYPE_CREDENTIAL_JSON,
+  TYPE_CREDENTIAL_USERNAME_PASSWORD_DOMAIN,
+} from 'api/models/credential';
+import usernamePasswordFormComponent from './username_password';
+import sshPrivateKeyFormComponent from './ssh_private_key';
+import jsonFormComponent from './json';
+import usernamePasswordDomainFormComponent from './username_password_domain';
+import { assert } from '@ember/debug';
+
+const modelTypeToComponent = {
```

Comment:
I think we'll also have the `password` type once lis' [PR](https://github.com/hashicorp/boundary-ui/pull/3067) makes it in